src/Core/Indexes/QuadTree.cs

```diff
@@ -0,0 +1,400 @@
+                this.contents.Add(geometry);
+            }
+
+            /// <summary>
+            /// Removes a geometry from the tree.
+            /// </summary>
+            /// <param name="geometry"> The geometry to be removed. </param>
+            /// <returns><c>true</c> if the removal was successful, otherwise <c>false</c>.</returns>
+            public Boolean Remove(IBasicGeometry geometry)
+            {
+                if (this.Contents.Remove(geometry))
+                    return true;
+
+                foreach (QuadTreeNode child in this.children)
+                {
+                    if (child.bound.Contains(geometry.Envelope))
+                    {
+                        return child.Remove(geometry);
+                    }
+                }
+
+                return false;
+            }
+
+            /// <summary>
+            /// Subdivides a nodes geometries between its children.
+            /// </summary>
+            private void SubdivideContents()
+            {
+                List<IBasicGeometry> markedForRemove = new List<IBasicGeometry>();
+
+                foreach (IBasicGeometry geometry in this.contents)
+                {
+                    foreach (QuadTreeNode child in this.children)
+                    {
+                        if (child.Envelope.Contains(geometry.Envelope))
+                        {
+                            child.Add(geometry);
+                            markedForRemove.Add(geometry);
+                        }
+                    }
+                }
+
+                foreach (IBasicGeometry geometry in markedForRemove)
+                    this.contents.Remove(geometry);
+            }
+
+            /// <summary>
+            /// Creates the children nodes of a node.
+            /// </summary>
+            private void CreateChildren()
+            {
+                this.children = new List<QuadTreeNode>();
+
+                double midX = this.bound.MinX + (this.bound.MaxX - this.bound.MinX) / 2;
+                double midY = this.bound.MinY + (this.bound.MaxY - this.bound.MinY) / 2;
+
+                this.children.Add(new QuadTreeNode(new Envelope(this.bound.MinX, midX, this.bound.MinY, midY)));
+                this.children.Add(new QuadTreeNode(new Envelope(midX, this.bound.MaxX, this.bound.MinY, midY)));
+                this.children.Add(new QuadTreeNode(new Envelope(this.bound.MinX, midX, midY, this.bound.MaxY)));
+                this.children.Add(new QuadTreeNode(new Envelope(midX, this.bound.MaxX, midY, this.bound.MaxY)));
+            }
+        }
+
+        /// <summary>
+        /// The root of the tree.
+        /// </summary>
+        private QuadTreeNode root;
+
+        /// <summary>
+        /// Gets a value indicating whether the index is read-only.
+        /// </summary>
+        /// <value><c>true</c> if the index is read-only; otherwise, <c>false</c>.</value>
+        public Boolean IsReadOnly => false;
```

Comment:
I suggest to be consistent on expression-bodied members, so either use them with all properties, or use the regular property format. (I myself prefer the regular fashion, but I'm probably just getting old.)